country-a-service/epps-service/src/main/java/dk/sundhedsdatastyrelsen/ncpeh/service/PrescriptionService.java

```diff
@@ -396,4 +364,112 @@ public GetDrugMedicationResponseType getDrugMedicationResponse(String cpr, List<
+            } catch (MapperException e) {
+                throw new DataRequirementException(String.format(MAPPING_ERROR_MESSAGE, e.getMessage()), e);
+            }
+        }
+
+        private ReqRes<StartEffectuationRequestType, StartEffectuationResponseType> lockEffectuation(
+            String patientId,
+            Document dispensationCda,
+            EuropeanHcpIdwsToken token
+        ) {
+            try {
+                var startEffectuationRequest = DispensationMapper.startEffectuationRequest(patientId, dispensationCda);
+                var startEffectuationResponse = fmkClient.startEffectuation(startEffectuationRequest, token);
+                return new ReqRes<>(startEffectuationRequest, startEffectuationResponse);
+            } catch (JAXBException e) {
+                throw new CountryAException(HttpStatus.INTERNAL_SERVER_ERROR, "StartEffectuation failed", e);
+            } catch (MapperException e) {
+                throw new DataRequirementException(String.format(MAPPING_ERROR_MESSAGE, e.getMessage()), e);
+            }
+        }
+
+        private void validateDispensable(String patientId, Document dispensationCda, EuropeanHcpIdwsToken token) {
+            try {
+                var prescriptionId = DispensationMapper.prescriptionId(dispensationCda);
+                var prescriptionResponse = fmkClient.getPrescription(
+                    GetPrescriptionRequestType.builder()
+                        .withPersonIdentifier().withSource("CPR").withValue(PatientIdMapper.toCpr(patientId)).end()
+                        .withIncludeOpenPrescriptions().end()
+                        .withIncludeEffectuations(true)
+                        .build(),
+                    token);
+                var prescription = prescriptionResponse.getPrescription()
+                    .stream()
+                    .filter(p -> p.getIdentifier() == prescriptionId)
+                    .findFirst()
+                    .orElseThrow(() -> new CountryAException(HttpStatus.NOT_FOUND, "Could not find prescription to dispense"));
+                var dispensationAllowedError = DispensationAllowed.getDispensationRestrictions(
+                    prescription,
+                    lmsDataProvider.packageInfo(prescription.getPackageRestriction()
+                        .getPackageNumber()
+                        .getValue()));
+                if (null != dispensationAllowedError) {
+                    throw new CountryAException(HttpStatus.BAD_REQUEST, "Prescription is not allowed to be dispensed. " + dispensationAllowedError);
+                }
+            } catch (XPathExpressionException | MapperException | JAXBException e) {
+                throw new CountryAException(HttpStatus.INTERNAL_SERVER_ERROR, "Could not fetch prescription to dispense", e);
+            }
+        }
+
+        private static String geteDispensationId(Document dispensationCda) {
```

Comment:
```suggestion
        private static String getEDispensationId(Document dispensationCda) {
```

would be the proper capitalisation, IMO